HazyResearch/snorkel
# RTD improvements

* Switch to autosummary from apidoc
  * Makes documentation much more legible
  * New page for every member, so more indexable
  * Requires creation of package page source files, handled by sync_api_docs.py
  * This has to be run manually and committed, however it is done by `tox -e fix` and checked by `tox -e check`
  * Currently documents everything importable via `from snorkel.XXX import YYY`, plus additional ones specified in packages.json (see #1358 for reasoning)
* Update Sphinx packages, simplify mocking
* Add logo to RTD, fix some paths, dynamic copyright
* Moves tbX as essential package, removes torchtext (unused)

Here's an initial build, ~~but there are issues with mocked packages~~: https://snorkel.readthedocs.io/en/rtd-nice/. compare to https://snorkel.readthedocs.io/en/redux/

~~Contacting RTD folks now to see if we can get them installed correctly.~~

**Test plan**
* Verify check functionality of sync_api_docs manually
* Verify tox commands manually
* Testing RTD compat as we go
